### PR TITLE
Addressing Some Xcode/Clang Warnings

### PR DESCRIPTION
**Pull request by sgowen**
Just a few implicit cast warnings and a strict-prototypes warning.

I didn’t build with CMake though, so its entirely possible that these warnings are being silenced and are therefore expected.

I just prefer to have as few warnings in my workspace as possible, even little ones like these if possible.

screen shot 2017-11-16 at 2 11 13 pm

### CODE REVIEW

File: src/input.c [hunk 1]
```diff
@@ -136,8 +136,8 @@ static GLFWbool parseMapping(_GLFWmapping* mapping, const char* string)
 
                 if (fields[i].element->type == _GLFW_JOYSTICK_HATBIT)
                 {
-                    const unsigned int hat = strtoul(c + 1, (char**) &c, 10);
-                    const unsigned int bit = strtoul(c + 1, (char**) &c, 10);
+                    const unsigned int hat = (unsigned int) strtoul(c + 1, (char**) &c, 10);
+                    const unsigned int bit = (unsigned int) strtoul(c + 1, (char**) &c, 10);
```
Comment (elmindreda) [inline src/input.c]: It's better to change the variable type to match here.

File: src/cocoa_window.m [hunk 2]
```diff
@@ -686,7 +686,7 @@ - (BOOL)performDragOperation:(id <NSDraggingInfo>)sender
                         [sender draggingLocation].x,
                         contentRect.size.height - [sender draggingLocation].y);
 
-    const int count = [files count];
+    const int count = (int) [files count];
```
Comment (elmindreda) [inline src/cocoa_window.m]: It's better to change `count` and `i` to `NSUInteger` here.

File: src/cocoa_joystick.m [hunk 3]
```diff
@@ -399,11 +399,13 @@ int _glfwPlatformPollJoystick(_GLFWjoystick* js, int mode)
 
             const long delta = axis->maximum - axis->minimum;
             if (delta == 0)
-                _glfwInputJoystickAxis(js, i, 0.f);
+            {
+                _glfwInputJoystickAxis(js, (int) i, 0.f);
+            }
```
Comment (elmindreda) [inline src/cocoa_joystick.m]: Please don't add braces for a single unbroken line.

Comment (sgowen) [reply src/cocoa_joystick.m]: Ah sorry, just force of habit.

**Comment by sgowen** [comment]
Awesome, made suggested changes, let me know how it looks!

**Comment by sgowen** [comment]
@elmindreda not entirely sure why the build is failing: https://ci.appveyor.com/project/elmindreda/glfw/build/1.0.646/job/w2oatgh3rm84oqn0

**Comment by elmindreda** [comment]
The cast has higher precedence than the bitwise operator, so the result gets promoted back up to `unsigned long`.

**Comment by sgowen** [comment]
Huzzah!

**Comment by sgowen** [comment]
@elmindreda GTM?

**Comment by elmindreda** [comment]
Thank you!

**Comment by sgowen** [comment]
:) and thank you for taking the time to review this and merge it in!

**Pull request closed without merging**